src/lib/services/client-metrics/instance-service.ts

```diff
@@ -162,8 +174,27 @@ export default class ClientInstanceService {
 
     async getApplications(
         query: IApplicationQuery,
+        userId: number,
     ): Promise<IClientApplication[]> {
-        return this.clientApplicationsStore.getAppsForStrategy(query);
+        const applications =
+            await this.clientApplicationsStore.getAppsForStrategy(query);
+        if (this.flagResolver.isEnabled('privateProjects') && userId) {
```

Comment:
Can userId be 0?

Reply:
That is good question. So far I have discovered we can have -1 userId, but have not seen 0.